### PR TITLE
#66 Get projectId by route

### PR DESCRIPTION
**Pull request by wanne27**

### CODE REVIEW

File: Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs [hunk 1]
```diff
@@ -19,8 +19,9 @@ public RolesPolicyHandler(IHttpContextAccessor httpContextAccessor)
 
         protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, RolesPolicyRequirement requirement)
         {
-            var queryProjectId = _httpContextAccessor.HttpContext.Request.Query.TryGetValue(Constants.Roles.ProjectIdParam, out var outProjectId);
-            var projectId = queryProjectId ? Guid.Parse(outProjectId) : DataAccess.Constants.GlobalProject.GlobalProjectId;
+            var queryProjectId = _httpContextAccessor.HttpContext.Request.Query.TryGetValue(Constants.Roles.ProjectIdParam, out var outQueryProjectId);
+            var routeProjectId = _httpContextAccessor.HttpContext.Request.RouteValues.TryGetValue(Constants.Roles.ProjectIdParam, out var outRouteProjectId);
+            var projectId = queryProjectId ? Guid.Parse(outQueryProjectId) : routeProjectId ? Guid.Parse(outRouteProjectId.ToString()) : DataAccess.Constants.GlobalProject.GlobalProjectId;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs:24]: why outRouteProjectId.ToString()? which type outRouteProjectId has?

File: Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs [hunk 2]
```diff
@@ -19,8 +19,9 @@ public RolesPolicyHandler(IHttpContextAccessor httpContextAccessor)
 
         protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, RolesPolicyRequirement requirement)
         {
-            var queryProjectId = _httpContextAccessor.HttpContext.Request.Query.TryGetValue(Constants.Roles.ProjectIdParam, out var outProjectId);
-            var projectId = queryProjectId ? Guid.Parse(outProjectId) : DataAccess.Constants.GlobalProject.GlobalProjectId;
+            var queryProjectId = _httpContextAccessor.HttpContext.Request.Query.TryGetValue(Constants.Roles.ProjectIdParam, out var outQueryProjectId);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs:22]: projectIdIsInQuery

File: Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs [hunk 3]
```diff
@@ -19,8 +19,9 @@ public RolesPolicyHandler(IHttpContextAccessor httpContextAccessor)
...
+            var routeProjectId = _httpContextAccessor.HttpContext.Request.RouteValues.TryGetValue(Constants.Roles.ProjectIdParam, out var outRouteProjectId);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs:23]: projectIdIsInRoute

File: Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs (hunk 1 above)
Comment (wanne27) [reply Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs:24]: object

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**